fuzzers/utils.py

```diff
@@ -150,21 +157,46 @@ def get_dictionary_path(target_binary):
     return None
 
 
-def set_default_optimization_flag(env=None):
-    """Set default optimization flag if none is already set."""
-    if not env:
+def set_fuzz_target(env=None):
+    """Set |FUZZ_TARGET| env flag."""
+    if env is None:
+        env = os.environ
+
+    env['FUZZ_TARGET'] = get_config_value('fuzz_target')
+
+
+def set_compilation_flags(env=None):
+    """Set compilation flags."""
+    if env is None:
         env = os.environ
 
-    for flag_var in ['CFLAGS', 'CXXFLAGS']:
-        append_flags(flag_var, [DEFAULT_OPTIMIZATION_LEVEL], env=env)
+    env['CFLAGS'] = ''
+    env['CXXFLAGS'] = ''
+
+    if get_config_value('type') == 'bug':
+        append_flags('CFLAGS',
+                     SANITIZER_FLAGS + [DEFAULT_OPTIMIZATION_LEVEL],
+                     env=env)
+        append_flags('CXXFLAGS',
+                     SANITIZER_FLAGS +
+                     [LIBCPLUSPLUS_FLAG, DEFAULT_OPTIMIZATION_LEVEL],
+                     env=env)
+    else:
+        append_flags('CFLAGS',
+                     NO_SANITIZER_COMPAT_CFLAGS + [DEFAULT_OPTIMIZATION_LEVEL],
+                     env=env)
+        append_flags('CXXFLAGS',
+                     NO_SANITIZER_COMPAT_CFLAGS +
+                     [LIBCPLUSPLUS_FLAG, DEFAULT_OPTIMIZATION_LEVEL],
+                     env=env)
 
 
 def initialize_flags(env=None):
```

Comment:
Maybe change this function name to `initialize_env` since it is initializing more than flags.